Makefile

```diff
@@ -33,82 +39,88 @@ test-final-private:
 
-test-e2e-phpdbg-70: vendor
-	docker run -it --rm -e PHPDBG=1 -v "$$PWD":/opt/infection -w /opt/infection php:7.0-cli ./tests/e2e_tests
+test-e2e-phpdbg-70: build-xdebug-70
+	$(DOCKER_RUN_70) ./tests/e2e_tests
 
-test-e2e-phpdbg-71: vendor
-	docker run -it --rm -e PHPDBG=1 -v "$$PWD":/opt/infection -w /opt/infection php:7.1-cli ./tests/e2e_tests
+test-e2e-phpdbg-71: build-xdebug-71
+	$(DOCKER_RUN_71) ./tests/e2e_tests
 
-test-e2e-phpdbg-72: vendor
-	docker run -it --rm -e PHPDBG=1 -v "$$PWD":/opt/infection -w /opt/infection php:7.2-cli ./tests/e2e_tests
+test-e2e-phpdbg-72: build-xdebug-72
+	$(DOCKER_RUN_72) ./tests/e2e_tests
 
 
 .PHONY: test-infection-xdebug test-infection-xdebug-70 test-infection-xdebug-71 test-infection-xdebug-72
 #infection with xdebug
 test-infection-xdebug: test-infection-xdebug-70 test-infection-xdebug-71 test-infection-xdebug-72
 
 test-infection-xdebug-70: build-xdebug-70
-	docker run -it --rm -v "$$PWD":/opt/infection -w /opt/infection infection_php70 php bin/infection --threads=4
+	$(DOCKER_RUN_70) php bin/infection --threads=4
 
 test-infection-xdebug-71: build-xdebug-71
-	docker run -it --rm -v "$$PWD":/opt/infection -w /opt/infection infection_php71 php bin/infection --threads=4
+	$(DOCKER_RUN_71) php bin/infection --threads=4
 
 test-infection-xdebug-72: build-xdebug-72
-	docker run -it --rm -v "$$PWD":/opt/infection -w /opt/infection infection_php72 php bin/infection --threads=4
+	$(DOCKER_RUN_72) php bin/infection --threads=4
 
 
 .PHONY: test-e2e-xdebug test-e2e-xdebug-70 test-e2e-xdebug-71 test-e2e-xdebug-72
 #e2e tests with xdebug
 test-e2e-xdebug: test-e2e-xdebug-70 test-e2e-xdebug-71 test-e2e-xdebug-72
 
 test-e2e-xdebug-70: build-xdebug-70
-	docker run -it --rm -v "$$PWD":/opt/infection -w /opt/infection infection_php70 ./tests/e2e_tests
+	$(DOCKER_RUN_70) ./tests/e2e_tests
 
-test-e2e-xdebug-71: build-xdebug-71
-	docker run -it --rm -v "$$PWD":/opt/infection -w /opt/infection infection_php71 ./tests/e2e_tests
+test-e2e-xdebug-71: vendor
+	$(DOCKER_RUN_71) ./tests/e2e_tests
 
-test-e2e-xdebug-72: build-xdebug-72
-	docker run -it --rm -v "$$PWD":/opt/infection -w /opt/infection infection_php72 ./tests/e2e_tests
+test-e2e-xdebug-72: vendor
+	$(DOCKER_RUN_72) ./tests/e2e_tests
 
 
 .PHONY: build-xdebug-70 build-xdebug-71 build-xdebug-72
 #Building images with xdebug
-build-xdebug-70: vendor
-	docker build -t infection_php70 -f "$$PWD/devTools/Dockerfile-php70-xdebug" .
-
-build-xdebug-71: vendor
-	docker build -t infection_php71 -f "$$PWD/devTools/Dockerfile-php71-xdebug" .
-
-build-xdebug-72: vendor
-	docker build -t infection_php72 -f "$$PWD/devTools/Dockerfile-php72-xdebug" .
+build-xdebug-70: vendor devTools/Dockerfile-php70-xdebug.json
+devTools/Dockerfile-php70-xdebug.json: devTools/Dockerfile-php70-xdebug
+	docker build -t infection_php70 -f devTools/Dockerfile-php70-xdebug .
+	docker image inspect infection_php70 >> devTools/Dockerfile-php70-xdebug.json
```

Comment:
Make needs something to pin on. Therefore these .json placeholders: if a source file changes, it'll trigger a new `docker build`

Reply:
Never knew you could do that :+1: